cpp/src/stream_compaction/drop_duplicates.cu

```diff
@@ -0,0 +1,303 @@
+              thrust::counting_iterator<IndexType>(0),
+              output, 
+              [first, comp, n] __device__ (IndexType i) mutable {
+              return (i == n || !comp(first[i], first[i+1]));
+              });
+  } else {
+      return thrust::copy_if(exec,
+              first,
+              last,
+              thrust::counting_iterator<IndexType>(0),
+              output, 
+              [first, comp, n] __device__ (IndexType i) mutable {
+              return (i == 0 || !comp(first[i], first[i-1])) 
+                  && (i == n || !comp(first[i], first[i+1]));
+              });
+  }
+}
+
+auto get_unique_ordered_indices(cudf::table_view const& keys,
+                                cudf::column &unique_indices,
+                                duplicate_keep_option const& keep,
+                                bool const& nulls_are_equal = true,
+                                rmm::mr::device_memory_resource* mr =
+                                rmm::mr::get_default_resource(),
+                                cudaStream_t stream=0)
+{
+  // sort only indices
+  auto sorted_indices = sorted_order(keys,
+                                     std::vector<order>{},
+                                     std::vector<null_order>{},
+                                     stream,
+                                     mr);
+
+
+  // extract unique indices 
+  auto device_input_table = cudf::table_device_view::create(keys, stream);
+
+  if(cudf::has_nulls(keys)) {
+    auto comp = row_equality_comparator<true>(*device_input_table,
+                                              *device_input_table,
+                                               nulls_are_equal);
+    auto result_end = unique_copy(rmm::exec_policy(stream)->on(stream),
+        sorted_indices->view().begin<cudf::size_type>(),
+        sorted_indices->view().end<cudf::size_type>(),
+        unique_indices.mutable_view().begin<cudf::size_type>(),
+        comp,
+        keep);
+  
+    return cudf::detail::slice(unique_indices.view(), 0, thrust::distance(unique_indices.mutable_view().begin<cudf::size_type>(), result_end));
+  } else {
+    auto comp = row_equality_comparator<false>(*device_input_table,
+                                               *device_input_table,
+                                               nulls_are_equal);
+    auto result_end = unique_copy(rmm::exec_policy(stream)->on(stream),
+        sorted_indices->view().begin<cudf::size_type>(),
+        sorted_indices->view().end<cudf::size_type>(),
+        unique_indices.mutable_view().begin<cudf::size_type>(),
+        comp,
+        keep);
+  
+    return cudf::detail::slice(unique_indices.view(), 0, thrust::distance(unique_indices.mutable_view().begin<cudf::size_type>(), result_end));
+  }
+  
+}
+
+cudf::size_type unique_count(table_view const& keys,
+                             bool const& nulls_are_equal = true,
+                             rmm::mr::device_memory_resource* mr 
```

Comment:
This API isn't used to return allocated memory, and therefore should _not_ accept a MR parameter. Temporary memory allocations should _always_ use the default resource.